src/main/java/com/example/bookstoreapp/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.example.bookstoreapp.repository.impl;
+
+import com.example.bookstoreapp.entity.Book;
+import com.example.bookstoreapp.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    private static final String CAN_T_SAVE_BOOK_MSG = "Can't save book";
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Autowired
+    public BookRepositoryImpl(EntityManagerFactory entityManagerFactory) {
+        this.entityManagerFactory = entityManagerFactory;
+    }
```

Comment:
the same. Autowired is not required here but I recommend replacing it with RequiredArgsConstructor annotation

Reply:
+ I set RequiredArgsConstructor in my Entity